adoc/chapters/architecture.adoc

```diff
@@ -306,22 +299,21 @@ across <<backend, SYCL backends>>.
 
 USM allocations are subject to the limitations described in <<sec:usm>>.
 
-When a SYCL application runs on any number of <<backend, SYCL backends>> without
-relying on any <<backend>>-specific behavior or interoperability, it is said to
-be a SYCL general application, and it is expected to run in any SYCL-conformant
+When a SYCL application runs without relying on any <<backend>>-specific
+behavior or interoperability, it is expected to run on any SYCL-conformant
 implementation that supports the required features for the application.
 
 
 === Platform mixed version support
 
-The SYCL generic programming model exposes a number of <<platform,platforms>>,
-each of them either empty or exposing a number of <<device,devices>>.
-Each <<platform>> is bound to a certain <<backend>>.
-SYCL <<device,devices>> associated with said <<platform>> are associated with
-that <<backend>>.
+The SYCL programming model exposes a number of <<platform,platforms>>, each of
+them either empty or exposing a number of <<device,devices>>.
+Each platform is bound to no more than one <<backend>>.
+If the platform is bound to a backend, the devices associated with that platform
+are associated with that backend.
 
-Although the APIs in the SYCL generic programming model are defined according to
-this specification and their version is indicated by the macro
+Although most APIs in the SYCL programming model are defined according to this
+specification and their version is indicated by the macro
```

Comment:
Even if it was in the previous version, `and their version` looks weird to me.
Perhaps we can clarify the meaning?

Reply:
If we want to clean this up, I would propose deleting this entire paragraph.  I don't think it adds any clarity.  I don't think the core spec needs to say anything about the versioning of the backend interoperation APIs.  This can be left to each backend interoperation specification.